queries/t32/highlights.scm

```diff
@@ -42,36 +55,125 @@
 [
   ","
   "."
-  ";"
 ] @punctuation.delimiter
 
 [
-  (access_class)
+  "enum"
+  "struct"
+  "union"
+] @keyword
+
+"sizeof" @keyword.operator
+
+[
+  "const"
+  "volatile"
+] @type.qualifier
+
+
+; Operators in comma and conditional HLL expressions
+(hll_comma_expression
+  "," @operator)
+
+(hll_conditional_expression
+  [
+   "?"
+   ":"
+] @conditional.ternary)
+
+
+; Strings and others literal types
+(access_class) @constant.builtin
+
+[
   (address)
   (bitmask)
   (file_handle)
-  (frequency)
   (integer)
+  (hll_number_literal)
+] @number
+
+[
+  (float)
+  (frequency)
   (percentage)
   (time)
-] @number
+] @float
 
-(float) @float
+[
+  (string)
+  (hll_string_literal)
+] @string
 
-(string) @string
+(hll_escape_sequence) @string.escape
 
 (path) @string.special
-
 (symbol) @symbol
 
-(character) @character
+[
+  (character)
+  (hll_char_literal)
+] @character
+
+
+; Types in HLL expressions
+[
+ (hll_type_identifier)
+ (hll_type_descriptor)
+] @type
+
+(hll_type_qualifier) @type.qualifier
+
+(hll_primitive_type) @type.builtin
+
+
+; HLL expressions
+(hll_call_expression
+  function: (identifier) @function.call)
+
+(hll_call_expression
+  function: (hll_field_expression
+    field: (hll_field_identifier) @function.call))
+
+(hll_call_expression
+  (hll_argument_list
+    (identifier) @variable))
+
+(hll_field_identifier) @field
+
+((hll_field_expression
+  (hll_field_identifier) @field)) @variable
+
+(hll_subscript_expression
+  argument: (identifier) @variable)
+
+(hll_pointer_expression
+  argument: (identifier) @variable)
+
+(hll_cast_expression
+  value: (identifier) @variable)
+
+(hll_binary_expression
+  (identifier) @variable)
+
+(hll_unary_expression
+  (identifier) @variable)
+
+(hll_update_expression
+  (identifier) @variable)
+
+(hll_assignment_expression
+  left: (identifier) @variable)
```

Comment:
With these many `(identifier) @variable` I think it's okay to have a `(identifier) @variable` instead of all of these. Not against narrowing here though, this will allow for less extmarks

Reply:
Amount of `(identifier) @variable` queries was reduced.